### DISCUSSION TITLE
Scoop doesn't install package properly. Error on executing: Run Setup.exe first!

### DISCUSSION THREAD

**Discussion by mattia72** [ScoopInstaller/Scoop#5393] [Help and Questions] [answered]
I tried to make an app manifest for mousewithoutborders like this:

    {
    "version": "2.2.1",
    "description": "Mouse without Borders (http://aka.ms/mm) is a product that makes you the captain of your computer fleet by allowing you to control up to four computers from a single mouse and keyboard. This means that with Mouse without Borders you can copy text or drag and drop files across computers.",
    "homepage": "http://aka.ms/mm",
    "license": "Public Domain",
    "url": "https://github.com/truong2d/mm/releases/download/2.2.1/MouseWithoutBordersSetup.msi",
    "hash": "6698dc23f33fea1330e383b9ca5b1699fae6418a8d89e48dc119cb76654f53b1",
    "extract_dir": "Microsoft Garage\\Mouse without Borders",
    "bin": "MouseWithoutBorders.exe",
    "shortcuts": [
        [
            "MouseWithoutBorders.exe",
            "Mouse without Borders"
        ]
    ],
    "checkver": {
        "github": "https://github.com/truong2d/mm"
    },
    "autoupdate": {
        "url": "https://github.com/truong2d/mm/releases/download/$version/MouseWithoutBordersSetup.msi"
    }
    }

The installation goes without any problem:

    > scoop install mousewithoutborders
    Installing 'mousewithoutborders' (2.2.1) [64bit] from my-scoop bucket
    Loading MouseWithoutBordersSetup.msi from cache.
    Checking hash of MouseWithoutBordersSetup.msi ... ok.
    Extracting MouseWithoutBordersSetup.msi ... done.
    Linking C:\home\mata\scoop\apps\mousewithoutborders\current => 
    C:\home\mata\scoop\apps\mousewithoutborders\2.2.1
    Creating shim for 'MouseWithoutBorders'.
    Creating shortcut for Mouse without Borders (MouseWithoutBorders.exe)
    'mousewithoutborders' (2.2.1) was installed successfully!

If I start the executable, I get the following error in a message box:

    Mouse without Borders is not installed yet, please run Setup.exe first!

Is there a way to avoid this error?

**Comment by rashil2000** [comment]
There's no Setup.exe file in the MSI package. You should contact the developer for this.

**Comment by r15ch13** [comment, marked as answer]
- Extract the msi file with `lessmsi`
- `scoop install ilspy`
- open `MouseWithoutBorders.exe` with `ILSpy`
- look for `Common.StartMouseWithoutBordersService()`

And there you have it: 
It requires the Service `MouseWithoutBordersSvc` to be installed. :smile:

![image](https://user-images.githubusercontent.com/432127/219970137-6242df58-d034-4d9a-afb5-b7c9cc472cf1.png)

**Reply by r15ch13** [reply]
To un/install the service run the following from an elevated console/powershell:

`C:\Windows\Microsoft.NET\Framework\v4.0.30319\InstallUtil.exe MouseWithoutBordersSvc.exe`
`C:\Windows\Microsoft.NET\Framework\v4.0.30319\InstallUtil.exe /u MouseWithoutBordersSvc.exe`

**Reply by rashil2000** [reply]
Wouldn't it be simpler to just use `sc create [...]`?

**Reply by mattia72** [reply]
I put this in the manifest file, and it works fine:

    "depends": ["gsudo"],
    "post_install" : [
      "gsudo sc create MouseWithoutBordersSvc binPath=$dir\\MouseWithoutBordersSvc.exe",
    ],
    "post_uninstall": [
      "gsudo sc delete MouseWithoutBordersSvc"
    ]


Thank you for your assistance
